### PR TITLE
🐛 source-mysql Support special chars in dbname

### PR DESCRIPTION
**Pull request by xiaohansong**
## What
https://github.com/airbytehq/airbyte-internal-issues/issues/2183
Currently we use dbName as topic.prefix in debezium property. However they have restrictions on special charaters and do not allow some special characters.

Problem exists in mssql and mysql. Postgres does not allow us to create db name with special characters so it's fine!

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumPropertiesManager.java [hunk 1]
```diff
@@ -76,15 +77,40 @@ public Properties getDebeziumProperties(
     props.setProperty("max.queue.size.in.bytes", BYTE_VALUE_256_MB);
 
     // WARNING : Never change the value of this otherwise all the connectors would start syncing from
-    // scratch
-    props.setProperty(TOPIC_PREFIX_KEY, getName(config));
+    // scratch.
```
Comment (akashkulk) [inline airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumPropertiesManager.java:80]: Are there any potential re-sync issues with existing syncs? I assume that because in these cases syncs are already in a broken state it shouldn't be an issue, but wanted to confirm

Comment (xiaohansong) [reply airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumPropertiesManager.java:80]: Yeah that's what I'm thinking too. The reformatting code is just borrowed from debezium sanitize code so I would expect them to recognize invalid name in the same way.

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java
Comment (akashkulk) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java:1]: Do we need to do the same for MsSqlDebeziumStateUtil

Comment (xiaohansong) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java:1]: yeah likely we need to, I'll try it out. And yeah I think it'd be easier to do it as a follow up fix

**Review by akashkulk — APPROVED** [review]
Minor comments - looks good in general

File: airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumPropertiesManager.java [hunk 2]
```diff
@@ -76,15 +77,40 @@ public Properties getDebeziumProperties(
...
+    props.setProperty(TOPIC_PREFIX_KEY, sanitizeTopicPrefix(getName(config)));
 
     // includes
     props.putAll(getIncludeConfiguration(catalog, config));
 
     return props;
   }
 
+  public static String sanitizeTopicPrefix(final String topicName) {
```
[the remark this answers is not in the document]
Comment (xiaohansong) [reply airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumPropertiesManager.java:89]: I borrowed it from the following and there is no good way to directly use them though: 

https://github.com/debezium/debezium/blob/c51ef3099a688efb41204702d3aa6d4722bb4825/debezium-core/src/main/java/io/debezium/schema/AbstractTopicNamingStrategy.java#L178

And I guess we want to keep them the same. I'll leave some breadcrumb in the comment for better readability.

[the remark this answers is not in the document]
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumPropertiesManager.java:89]: @akashkulk I do prefer the replaceAll as well. It's simpler, and we'll use it infrequently enough that perf don't matter
But with a correct one : "[^a-zA-Z0-9._-]" ('.' loses its special meaning inside square brackets and '-' needs to be at the end)

File: airbyte-cdk/java/airbyte-cdk/db-sources/src/testFixtures/java/io/airbyte/cdk/integrations/debezium/CdcSourceTest.java [hunk 3]
```diff
@@ -84,11 +84,11 @@ public abstract class CdcSourceTest<S extends Source, T extends TestDatabase<?,
   protected T testdb;
 
   protected String createTableSqlFmt() {
-    return "CREATE TABLE %s.%s(%s);";
+    return "CREATE TABLE `%s`.`%s`(%s);";
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/db-sources/src/testFixtures/java/io/airbyte/cdk/integrations/debezium/CdcSourceTest.java:87]: uh? back ticks? I doubt this is an universally working change. If it is, it probably deserve a comment with a reference.

Comment (xiaohansong) [reply airbyte-cdk/java/airbyte-cdk/db-sources/src/testFixtures/java/io/airbyte/cdk/integrations/debezium/CdcSourceTest.java]: yeah i think you are right. I'll push this change to a lower level and apply it only to mysql

File: airbyte-integrations/connectors/source-mysql/src/testFixtures/java/io/airbyte/integrations/source/mysql/MySQLTestDatabase.java [hunk 4]
```diff
@@ -70,13 +79,34 @@ public MySQLTestDatabase withoutStrictMode() {
...
+  }
 
   @Override
   protected Stream<Stream<String>> inContainerBootstrapCmd() {
+    System.out.println("name: " + String.format("CREATE DATABASE `%s`", getDatabaseName()));
     return Stream.of(mysqlCmd(Stream.of(
         String.format("SET GLOBAL max_connections=%d", MAX_CONNECTIONS),
-        String.format("CREATE DATABASE %s", getDatabaseName()),
-        String.format("CREATE USER '%s' IDENTIFIED BY '%s'", getUserName(), getPassword()),
+        String.format("CREATE DATABASE \\`%s\\`", getDatabaseName()),
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/testFixtures/java/io/airbyte/integrations/source/mysql/MySQLTestDatabase.java:108]: Why this change in a test class? Why the backticks? Why the change bellow to the `CREATE USER`
Are we adding a test case for this?
